runner/src/main.rs

```diff
@@ -279,120 +392,292 @@ fn run_example_server(opt: &BuildServer, application_file: &str) -> Cmd {
+                    ],
+                ),
+            },
+            Step::Single {
+                name: "build example image".to_string(),
+                command: Cmd::new(
+                    "docker",
+                    &[
+                        "build",
+                        &format!("--tag={}", example.name),
+                        "--file=./examples/Dockerfile",
+                        &format!("./examples/{}", example.name),
+                    ],
+                ),
+            },
+        ],
+    }
+}
+
+fn build(target: &Target) -> Box<dyn Runnable> {
+    match target {
+        Target::Cargo { cargo_manifest } => Cmd::new(
+            "cargo",
+            vec![
+                "build".to_string(),
+                "--release".to_string(),
+                format!("--target={}", DEFAULT_EXAMPLE_BACKEND_RUST_TARGET),
+                format!("--manifest-path={}", cargo_manifest),
+            ],
+        ),
+        Target::Bazel {
+            bazel_target,
+            config,
+        } => Cmd::new(
+            "bazel",
+            spread![
+                "build".to_string(),
+                ...if config.is_empty() {
+                    vec![]
+                } else {
+                    vec![format!("--config={}", config)]
                 },
-                Step::Single {
-                    name: "npm start".to_string(),
-                    command: Cmd::new(
-                        "npm",
-                        vec![
-                            "start".to_string(),
-                            "--prefix".to_string(),
-                            package_directory.to_string(),
-                        ],
-                    ),
+                bazel_target.to_string(),
+            ],
+        ),
+        Target::Npm { package_directory } => Cmd::new(
+            "npm",
+            vec!["ci".to_string(), format!("--prefix={}", package_directory)],
+        ),
+        Target::Shell { script } => Cmd::new("bash", &[script]),
+    }
+}
+
+fn run(executable: &Executable, additional_args: Vec<String>) -> Box<dyn Runnable> {
+    match &executable.target {
+        Target::Cargo { cargo_manifest } => Cmd::new(
+            "cargo",
+            spread![
+                "run".to_string(),
+                "--release".to_string(),
+                format!("--target={}", DEFAULT_EXAMPLE_BACKEND_RUST_TARGET),
+                format!("--manifest-path={}", cargo_manifest),
+                "--".to_string(),
+                ...executable.additional_args.clone(),
+                ...additional_args,
+            ],
+        ),
+        Target::Bazel {
+            bazel_target,
+            config,
+        } => Cmd::new(
+            "bazel",
+            spread![
+                "run".to_string(),
+                ...if config.is_empty() {
+                    vec![]
+                } else {
+                    vec![format!("--config={}", config)]
                 },
```

Comment:
conditional spreading using an empty vec :p. 

I like it, though in the past my colleagues always found it confusing when I did the JS equivalent. 🙊

```js
let foo = [
   // ...
   ...(shouldSpread ? array : []
]
```